main.go

```diff
@@ -56,6 +56,16 @@ func run(ctx context.Context) {
 			logger.Errorf(err.Error())
 		}
 
+		if _, err := os.Stat(config.RestartFile()); err == nil {
+			logger.Infof("Restart signal recieved, waiting for tasks to complete.")
```

Comment:
think this should indicate more explicitly what's going on - no signal has actually been received, a marker file was detected.